cmd/kwil-admin/nodecfg/toml.go

```diff
@@ -257,4 +273,34 @@ max_tx_bytes = {{ .ChainCfg.Mempool.MaxTxBytes }}
 
 # Size of the cache (used to filter transactions we saw earlier) in transactions
 cache_size = {{ .ChainCfg.Mempool.CacheSize }}
+
+#######################################################
+###         State Sync Configuration Options        ###
+#######################################################
+[chain.statesync]
+# State sync rapidly bootstraps a new node by discovering, fetching, and restoring a state machine
+# snapshot from peers instead of fetching and replaying historical blocks. Requires some peers in
+# the network to take and serve state machine snapshots. State sync is not attempted if the node
+# has any local state (LastBlockHeight > 0). The node will have a truncated block history,
+# starting from the height of the snapshot.
+enable = {{ .ChainCfg.StateSync.Enable }}
+
+# Trusted snapshot providers (comma-separated chain RPC servers) are the source-of-truth for the snapshot integrity.
+# Snapshots are accepted for statesync only after verifying it with these trusted snapshot providers.
+# These are also used for light client verification of the synced state machine and
+# retrieval of state data for node bootstrapping.
+# Light client verification needs a trusted height and block hash at the configured block height obtained from a
+# trusted source, and a period during which validators can be trusted.
+# Atleast 2 trusted rpc servers are required for enabling state sync.
+rpc_servers = "{{ .ChainCfg.StateSync.RPCServers }}"
+trust_height = {{ .ChainCfg.StateSync.TrustHeight }}
+trust_hash = "{{ .ChainCfg.StateSync.TrustHash }}"
+trust_period = "{{ .ChainCfg.StateSync.TrustPeriod }}"
+
+# Time to spend discovering snapshots before initiating a restore.
+discovery_time = "{{ .ChainCfg.StateSync.DiscoveryTime }}"
```

Comment:
What are the considerations in setting `discovery_time`.  Looks like the default is 15 seconds, but that seems very short if your node has to use PEX for any amount of time to reach nodes that are serving the snap.

What happens if it's longer?  Does that make it annoying if there aren't any snap-serving nodes?  You have the default of `Enable: false` (sensibly), so it seems a longer discover time would be fine if they explicitly turn it on.